### ISSUE TITLE
expo upload:android throws SyntaxError parsing JSON

### ISSUE THREAD

**Issue by mymattcarroll** [expo/expo-cli#2321]
## Description

`expo upload:android` is throwing `SyntaxError` parsing JSON at the end of the command. As you can see from the output pasted below, the command actually succeeds and uploads to Google Play Store but then throws an error at the end. I can verify that the `.aab` file is successfully uploaded and published to Google Play production track :).

## Expected Behavior

Should not throw `SyntaxError`

## Observed Behavior

```
[17:33:21] Downloading your app archive

┌─────────────────────────────────────────────────────────────────────────────────┐
│                           Android Submission Summary                            │
├─────────────────────────┬───────────────────────────────────────────────────────┤
│ Android package         │ com.company.domain01.appnamehere                      │
├─────────────────────────┼───────────────────────────────────────────────────────┤
│ Archive path            │ /var/folders/cf/2l8j0rc11n3dr360_yybtzt80000gn/T/c    │
│                         │ om-company-domain01-appnamehere-47e867c7b2c843bdbf    │
│                         │ aad71b999d935e-signed.aab                             │
├─────────────────────────┼───────────────────────────────────────────────────────┤
│ Archive type            │ aab                                                   │
├─────────────────────────┼───────────────────────────────────────────────────────┤
│ Google Service Account  │ /Users/user/path/here/here/google-play.test.json      │
├─────────────────────────┼───────────────────────────────────────────────────────┤
│ Release track           │ production                                            │
├─────────────────────────┼───────────────────────────────────────────────────────┤
│ Release status          │ completed                                             │
├─────────────────────────┼───────────────────────────────────────────────────────┤
│ Submission mode         │ Submitting the app from this computer                 │
└─────────────────────────┴───────────────────────────────────────────────────────┘

+-------------------------+----------------------------------------------------------------------------+
|                                      Summary for supply 2.146.1                                      |
+-------------------------+----------------------------------------------------------------------------+
| package_name            | com.company.domain01.appnamehere                                           |
| json_key                | /Users/user/path/here/here/google-play.test.json                           |
| track                   | production                                                                 |
| aab                     | /var/folders/cf/2l8j0rc11n3dr360_yybtzt80000gn/T/com-company-domain01-     |
|                         | appnamehere-47e867c7b2c843bdbfaad71b999d935e-signed.aab                    │
| release_status          | completed                                                                  |
| skip_upload_apk         | false                                                                      |
| skip_upload_aab         | false                                                                      |
| skip_upload_metadata    | false                                                                      |
| skip_upload_changelogs  | false                                                                      |
| skip_upload_images      | false                                                                      |
| skip_upload_screenshots | false                                                                      |
| validate_only           | false                                                                      |
| check_superseded_tracks | false                                                                      |
| timeout                 | 300                                                                        |
| deactivate_on_promote   | true                                                                       |
+-------------------------+----------------------------------------------------------------------------+

[17:33:46]: Preparing aab at path '/var/folders/cf/2l8j0rc11n3dr360_yybtzt80000gn/T/com-company-domain01-appnamehere-47e867c7b2c843bdbfaad71b999d935e-signed.aab' for upload...

[17:35:40]: Updating track 'production'...

[17:35:42]: Uploading all changes to Google Play...

[17:35:54]: Successfully finished the upload to Google Play

[17:35:54] Unexpected token N in JSON at position 0

[17:35:54] SyntaxError: Unexpected token N in JSON at position 0
    at JSON.parse (<anonymous>)
    at runTravelingFastlaneAsync (/Users/user/path/here/here/node_modules/expo-cli/src/commands/upload/submission-service/utils/travelingFastlane.ts:18:20)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at AndroidOfflineSubmitter.submitAsync (/Users/user/path/here/here/node_modules/expo-cli/src/commands/upload/submission-service/android/AndroidSubmitter.ts:143:7)
    at AndroidSubmitter.submitOfflineAsync (/Users/user/path/here/here/node_modules/expo-cli/src/commands/upload/submission-service/android/AndroidSubmitter.ts:81:5)
    at AndroidSubmitter.submitAsync (/Users/user/path/here/here/node_modules/expo-cli/src/commands/upload/submission-service/android/AndroidSubmitter.ts:53:7)
    at AndroidSubmitCommand.runAsync (/Users/user/path/here/here/node_modules/expo-cli/src/commands/upload/submission-service/android/AndroidSubmitCommand.ts:45:5)
    at /Users/user/path/here/here/node_modules/expo-cli/src/commands/upload.ts:56:7
    at Command.<anonymous> (/Users/user/path/here/here/node_modules/expo-cli/src/exp.ts:80:7)
```

## Environment

```
Expo CLI 3.21.13 environment info:
    System:
      OS: macOS 10.15.5
      Shell: 5.5.1 - /usr/local/bin/zsh
    Binaries:
      Node: 12.14.1 - /usr/local/bin/node
      Yarn: 1.10.1 - /usr/local/bin/yarn
      npm: 6.14.5 - ~/.npm-packages/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Xcode: 11.5/11E608c - /usr/bin/xcodebuild
    npmPackages:
      expo: ^37.0.0 => 37.0.7 
      react: 16.9.0 => 16.9.0 
      react-native: https://github.com/expo/react-native/archive/sdk-37.0.0.tar.gz => 0.61.4 
      react-navigation: 4.0.10 => 4.0.10 
```

## Reproducible Demo

As this is an upload command, I dont see the point in making a snack to reproduce this error. However if there is more information that can help, please let me know.

**Comment by mymattcarroll** [comment]
reverting back to `expo-cli@3.17.24` fixes the issue.

**Comment by dsokal** [comment]
Hello @mymattcarroll,
Could you please find `/Users/user/path/here/here/node_modules/expo-cli/build/commands/upload/submission-service/utils/travelingFastlane.js`, add `console.log(stderr);` just before `JSON.parse(stderr)`, run `expo upload:android` again and share the log output? It'd help to debug this issue a lot. 
Thanks!

**Comment by mymattcarroll** [comment]
Re-creating now. I am slightly worried that you cannot easily reproduce this issue. Is this potentially something we have setup incorrectly in our project?

**Comment by mymattcarroll** [comment]
I don't seem to be able to replicate this issue! 😠 

The `console.log(stderr);` line is logging:

```sh
{"result":"success"}
```

This can be closed it seems 😄

**Comment by mymattcarroll** [comment]
Its happening again...😢

**Comment by brentvatne** [comment]
can you try the `--use-submission-service` flag? `expo upload:android --use-submission-service`

**Comment by mymattcarroll** [comment]
Trying now.

**Comment by mymattcarroll** [comment]
Perfection!!! Thank you @brentvatne 👍 

Will the `--use-submission-service` flag be added to the `expo upload:ios` command soon?

**Comment by brentvatne** [comment]
we'll work on it in the near future but we just have some higher priority projects right now

**Comment by mymattcarroll** [comment]
Understandable.

Is there an ETA on when the flag will no longer be "Experimental"?

**Comment by cannc4** [comment]
Hey @brentvatne I am experiencing same issue when I run
```
expo upload:ios --apple-id $APPLE_ID --apple-id-password $APPLE_PASSWORD 
```

any thoughts?

**Comment by mymattcarroll** [comment]
@cannc4 This would fix the issue: #2530
